File: Sliding_Window_18_Nov_2023/1695_maximum_erasure_value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        counter = [0] * 10001
        start, end, subarray_sum, _sum = 0, 0, 0, 0

        while end < len(nums):
            counter[nums[end]] += 1
            _sum += nums[end]

            while counter[nums[end]] > 1:
                counter[nums[start]] -= 1
                _sum -= nums[start]
                start += 1

            subarray_sum = max(subarray_sum, _sum)
            end += 1

        return subarray_sum

File: Sliding_Window_18_Nov_2023/1695-maximum-erasure-value.go
package main

func maximumUniqueSubarray(nums []int) int {
    counter := [10001]int{}
    start, end, subArraySum, tempSum := 0, 0, 0, 0

    for end < len(nums) {
        counter[nums[end]]++
        tempSum += nums[end]

        for counter[nums[end]] > 1 {
            counter[nums[start]]--
            tempSum -= nums[start]
            start++
        }

        subArraySum = max(subArraySum, tempSum)
        end++
    }

    return subArraySum
}
